File: Sunrin-ICPC-2022/Final-G/FG.java
import java.util.*;

public class Main{
    public static void main(String[] argv){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int q = sc.nextInt();
        long[] a = new long[n+2], x = new long[n+2];
        long[][] w = new long[m+1][n+2];

        for(int i=1; i<=m; i++){
            int cnt = sc.nextInt();
            int[] idx = new int[cnt+1];
            for(int j=0; j<cnt; j++) idx[j] = sc.nextInt();
            idx[cnt] = n + 1;
            for(int j=0; j<cnt+1; j++) w[i][idx[j]] = sc.nextInt();
        }
        for(int i=1; i<=m; i++){
            int t = sc.nextInt();
            for(int j=1; j<=n+1; j++) a[j] += t * w[i][j];
        }
        int t = sc.nextInt(); a[n+1] += t;

        for(int iter=1; iter<=q; iter++){
            for(int i=1; i<=n; i++) x[i] = sc.nextInt();
            x[n+1] = 1;
            long res = 0;
            for(int i=1; i<=n+1; i++) res += a[i] * x[i];
            System.out.println(res);
        }
    }
}

File: Sunrin-ICPC-2022/Final-G/FG.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

ll N, M, Q, A[5050], W[5050][5050], X[5050];

int main(){
    ios_base::sync_with_stdio(false); cin.tie(nullptr);
    cin >> N >> M >> Q;
    for(int i=1; i<=M; i++){
        int cnt; cin >> cnt;
        vector<int> idx(cnt);
        for(auto &j : idx) cin >> j;
        idx.push_back(N+1);
        for(int j=0; j<idx.size(); j++) cin >> W[i][idx[j]];
    }
    for(int i=1; i<=M; i++){
        ll t; cin >> t;
        for(int j=1; j<=N+1; j++) A[j] += t * W[i][j];
    }
    ll t; cin >> t; A[N+1] += t;

    for(int q=1; q<=Q; q++){
        for(int i=1; i<=N; i++) cin >> X[i]; X[N+1] = 1;
        ll res = 0;
        for(int i=1; i<=N+1; i++) res += A[i] * X[i];
        cout << res << "\n";
    }
}